motivaa-backend/motivaa-app/src/main/java/com/motivaa/ReniReview/SkyeIndexManager.java

```diff
@@ -0,0 +1,135 @@
+
+    public static void uploadIndexIfDoesntExists (
+            String indexName,
+            String filePath,
+            String endpoint
+    ) {
+        String emptyIndex   Version = "";
+        uploadIndexIfDoesntExists(
+                indexName,
+                emptyIndexVersion,
+                filePath,
+                endpoint
+        );
+    }
+
+    public static void uploadIndexIfDoesntExists (
+            String indexName,
+            String indexVersion,
+            String filePath,
+            String endpoint
+    ) {
+        boolean doesIndexExistOnStage = verifyIndexExistsOnStage(
+                indexName,
+                indexVersion,
+                endpoint
+        );
+
+        if ( doesIndexExistOnStage ) {
+            logger.info("Index '" + indexName + "' with version '" + indexVersion + "' already exists.");
+        }
+        else {
+            uploadCsvToSkyeAsIndex(
+                    indexName,
+                    indexVersion,
+                    filePath,
+                    endpoint
+            );
+            logger.info("Index '" + indexName + "' wutg version '" + indexVersion + "'  successfully created.");
+        }
+    }
+
+    private static boolean verifyIndexExistsOnStage(
+            String indexName,
+            String indexVersion,
+            String endpoint) {
+
+        Map<String, Object> requestBody = new HashMap<>();
+        requestBody.put("indexName", indexName);
+        requestBody.put("pageSize", "1");
+        requestBody.put("pageNumber", "1");
+        if ( !indexVersion.isEmpty() ) {
+            requestBody.put("indexVersion",indexVersion);
+        }
+
+        Response response = RestAssured
+                .given()
+                    .body(requestBody)
+                    .post(endpoint)
+                .then()
+                    .extract().response();
+        return response.getStatusCode() == 200;
+    }
+
+    private static void uploadCsvToSkyeAsIndex(
+            String indexName,
+            String indexVersion,
+            String filePath,
+            String endpoint
+    ) {
+
+        Map<String, Object> requestBody = prepareRequestBodyFromCsv(
+                indexName,
+                indexVersion,
+                filePath
+        );
+
+        sendCsvToSkye(
+                endpoint,
+                requestBody
+        );
+    }
+
+    private static Map<String,Object> prepareRequestBodyFromCsv(
+            String indexName,
+            String indexVersion,
+            String filePath
+    ) {
+        byte[] fileContentBytes;
+        try {
+            fileContentBytes = Files.readAllBytes(Paths.get(filePath));
+        } catch ( IOException ioException) {
+            logger.error("Error while reading the file from: \" + filePath",ioException);
+            throw new IndexUploadFailedException("Error while reading the file from: " + filePath);
```

Comment:
I think you should have this error string decplared before the log and use it in both the error log and the throw, so you don't have the same string twice.